presto-common/src/main/java/com/facebook/presto/common/type/DistinctTypeInfo.java

```diff
@@ -105,6 +105,21 @@ public boolean equals(Object o)
         return Objects.equals(name, other.name);
     }
 
+    public boolean equalsExact(Object o)
```

Comment:
Curious why we need to compare everything beyond `name`? Kinda have a hard time to understand the logic and its callsites

Reply:
This is for parametricTypesCache in BuiltInTypeAndFunctionManager. It can cache Array<DistinctType(a)> based on just name of distinct type, but later when we try to do getType(Array<DistincTypeSignature(a, parents = [b])>, it will give us the old cached type without any parent info